homeassistant/components/skybell/config_flow.py

```diff
@@ -0,0 +1,86 @@
+
+from .const import DEFAULT_CACHEDB, DOMAIN
+
+
+class SkybellFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Skybell."""
+
+    VERSION = 1
+
+    async def async_step_import(self, user_input: ConfigType) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input: dict[str, str] = None) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            email = user_input[CONF_EMAIL].lower()
+            password = user_input[CONF_PASSWORD]
+
+            if self.context["source"] != config_entries.SOURCE_REAUTH:
+                self._async_abort_entries_match({CONF_EMAIL: email})
+            user_id, error = await self._async_validate_input(email, password)
+            if error is None:
+                entry = await self.async_set_unique_id(user_id)
+                if entry:
+                    self.hass.config_entries.async_update_entry(entry, data=user_input)
+                    await self.hass.config_entries.async_reload(entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=email,
+                    data={CONF_EMAIL: email, CONF_PASSWORD: password},
+                )
+            errors["base"] = error
+
+        user_input = user_input or {}
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_EMAIL, default=user_input.get(CONF_EMAIL)): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def _async_validate_input(self, email: str, password: str) -> tuple:
+        """Validate login credentials."""
+        skybell = Skybell(
+            email,
+            password,
+            cache_path=self.hass.config.path(DEFAULT_CACHEDB),
+            disable_cache=True,
+            session=async_get_clientsession(self.hass),
+        )
+        try:
+            devices = await skybell.async_get_devices()
+        except exceptions.SkybellAuthenticationException:
+            return None, "invalid_auth"
+        except exceptions.SkybellException:
+            return None, "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            return None, "unknown"
+        return devices[0].user_id, None
+
+    async def async_step_reauth(self, config: dict[str, Any]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        return await self.async_step_user()
```

Comment:
Please leave reauth for another turn.  We need to check to make sure they don't switch accounts here and its better to simply return False in the `async_setup_entry` to keep this from getting any bigger for now